tests/unit/models/project-test.js

```diff
@@ -461,6 +461,16 @@ describe('models/project.js', function() {
       var addon = project.findAddonByName('not-an-addon');
       expect(addon).to.equal(undefined, 'not found addon should be undefined');
     });
+
+    it('should not return an addon that is a substring of requested name', function() {
+      var addon = project.findAddonByName('foo-ba');
+      expect(addon).to.equal(undefined, 'foo-ba should not be found');
+    })
+
+    it('should return an addon that is a suffix of requested name', function() {
+      var addon = project.findAddonByName('bar');
+      expect(addon).to.equal(undefined, 'should have found the foo-bar addon');
```

Comment:
I was thinking about that, but wasn't sure if that was desirable, but I sure can do that if that is more wanted.